csrc/scheduler/matmul.cpp

```diff
@@ -614,47 +617,96 @@ void scheduleOutputTensor(
+      [&epilogue_inputs](TensorView* tv) {
+        return (
+            std::find(epilogue_inputs.cbegin(), epilogue_inputs.cend(), tv) !=
+            epilogue_inputs.cend());
+      });
+
+  // No input tensor can be vectorized so all fusion inputs will be
+  //  parallized without vectorization mode propagated.
+  if (epilogue_input_tvs_with_vec.empty()) {
+    return {epilogue_input_tvs_with_vec, epilogue_inputs};
+  }
+
+  std::set<TensorView*> sink(
+      epilogue_input_tvs_with_vec.begin(), epilogue_input_tvs_with_vec.end());
+  std::vector<TensorView*> epilogue_input_tvs_without_vec = {};
+
+  for (auto* tv : epilogue_inputs) {
+    const auto status = sink.insert(tv);
+    if (status.second) {
+      epilogue_input_tvs_without_vec.push_back(tv);
+    }
+  }
+
+  return {epilogue_input_tvs_with_vec, epilogue_input_tvs_without_vec};
+}
+
 //! Propagates transformations from fusion output to fusion tv inputs that are
 //!  producers in the epilogue. Transformations' propagation aims at input tvs
 //!  which are not assigned to core roles, that is, are not MMA inputs.
-void scheduleFusionInputsForEpilogue(
-    const mma_utils::RolesMap& roles_map,
-    const bool with_smem_epilogue) {
-  std::vector<TensorView*> cached_tvs;
-
+void scheduleFusionInputsForEpilogue(const mma_utils::RolesMap& roles_map) {
   // Handling transformations in fusion input tvs with assigned INPUT_C role by
-  //  propagating fusion output transformations through cached views of INPUT_C
-  //  fusion input tvs and by setting vectorization of the inner most iterdomain
-  //  of these cached views
+  //  propagating fusion output transformations and parallelization through
+  //  cached views of fusion inputs with INPUT_C role.
+  // NOTE: Vectorization parallelization mode of inner most domain requires
+  //  special care.
   if (roles_map.count(MatmulRole::INPUT_C)) {
-    auto& c_tvs = roles_map.at(MatmulRole::INPUT_C);
-
     // The system supports only scenario where there is only one fusion output
-    //  with assigned OUTPUT_D role, this condition is already verified so there
-    //  is no need for an additional checks here
+    //  with assigned OUTPUT_D role, this condition is already verified so no
+    //  additional checks are needed
     auto output_d = roles_map.at(MatmulRole::OUTPUT_D).front();
-    for (auto* c : c_tvs) {
-      cached_tvs.push_back(c->cacheAfter());
+    const auto& c_tvs = roles_map.at(MatmulRole::INPUT_C);
+
+    const auto [tvs_with_vectorization, tvs_without_vectorization] =
+        partitionEpilogueInputsByParallelizationModes(output_d, c_tvs);
+
+    const auto transform_propagator =
+        [&output_d](
+            const std::vector<TensorView*>& tvs,
+            const std::unordered_set<ParallelType>& parallel_types) {
+          std::vector<TensorView*> cached_tvs;
+          cached_tvs.reserve(tvs.size());
+          for (auto* tv : tvs) {
+            cached_tvs.push_back(tv->cacheAfter());
```

Comment:
`emplace_back`

Reply:
I'm not convinced about using `emplace_back` as it usually suggest that we construct new object with attributes accepted by object class constructors, while `push_back` simply adds object to the collection. In this case we are pushing there pointers, so no new object is constructed.

Both calls will work, it is only a matter of expressing the intent.